tests/conftest.py

```diff
@@ -0,0 +1,511 @@
+        conditions=MappingProxyType({"stimulus": [-1, 1], "action": [-1, 1]}),
+        data=data,
+        key="base",
+        num_neurons=data.get("raster").shape[-1],
+        num_trials=len(set(data.get("trial").tolist())),
+        num_samples=data.get("raster").shape[0],
+        num_conditions=2,
+    )
+
+
+# noinspection DuplicatedCode
+@DataRegistry.register()
+def undersampled_data() -> Data:
+    """
+    An undersampled dataset for testing.
+    """
+    data = generate_synthetic_data(n_neurons=80,
+                                   n_trials=8,
+                                   keyA="stimulus",
+                                   rateA=0.3,
+                                   keyB="action",
+                                   rateB=0.3,
+                                   timebins_per_trial=1,
+                                   corrAB=0.0,
+                                   scale=1.0,
+                                   meanfr=0.1,
+                                   mixed_term=0.0,
+                                   mixing_factor=0.0
+                                   )
+    return Data(
+        conditions=MappingProxyType({"stimulus": [-1, 1], "action": [-1, 1]}),
+        data=data,
+        key="undersampled",
+        num_neurons=data.get("raster").shape[-1],
+        num_trials=len(set(data.get("trial").tolist())),
+        num_samples=data.get("raster").shape[0],
+        num_conditions=2,
+    )
+
+
+# noinspection DuplicatedCode
+@DataRegistry.register()
+def tangled_data() -> Data:
+    """
+    A dataset with tangled conditions for testing.
+    """
+    data = generate_synthetic_data(n_neurons=80,
+                                   n_trials=100,
+                                   keyA="stimulus",
+                                   rateA=0.3,
+                                   keyB="action",
+                                   rateB=0.0,
+                                   timebins_per_trial=5,
+                                   corrAB=0.8,
+                                   scale=1.0,
+                                   meanfr=0.1,
+                                   mixed_term=0.0,
+                                   mixing_factor=0.0
+                                   )
+    return Data(
+        conditions=MappingProxyType({"stimulus": [-1, 1], "action": [-1, 1]}),
+        data=data,
+        key="tangled",
+        num_neurons=data.get("raster").shape[-1],
+        num_trials=len(set(data.get("trial").tolist())),
+        num_samples=data.get("raster").shape[0],
+        num_conditions=2,
+    )
+
+
+"""
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Test Cases (Decoders)
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+"""
+
+
+# noinspection DuplicatedCode
+@DecodandaRegistry.register()
+def base_decoder(data: Data) -> TrainedDecodanda:
```

Comment:
Please rename all the "decoder" to "decodanda" for consistency (they inherit from the class TrainedDecodanda)

Reply:
Removed references to `decoder` as discussed. Classifier implementations are now provided as `classifiers